tests/main.go

```diff
@@ -311,6 +314,13 @@ func main() {
 			exitCode++
 		}
 	}
+
+	for key := range expected {
+		if !seen[key] {
+			fmt.Printf("❌ Expected output '%s' was not present\n", key)
+			exitCode++
```

Comment:
This is an existing pattern already with doing just `exitCode++`, but it strikes me as weird. I guess the intention is to count the number of errors, but e.g. on Linux, if we end up with an `os.Exit(256)`, we effectively `exit 0`, and the failure wouldn't be noticed in CI.  Not that there's a high risk of that, but I'd feel better just doing `exitCode = 1` here instead.

Reply:
I'll do that while I'm here -- good shout! Seems there are some new tests I need to fix up too. Thanks!